bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/command/NanoleafCommandExtension.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nanoleaf.internal.command;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.nanoleaf.internal.NanoleafBindingConstants;
+import org.openhab.binding.nanoleaf.internal.handler.NanoleafControllerHandler;
+import org.openhab.core.io.console.Console;
+import org.openhab.core.io.console.extensions.AbstractConsoleCommandExtension;
+import org.openhab.core.io.console.extensions.ConsoleCommandExtension;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingRegistry;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * Console commands for interacting with Nanoleaf integration
+ *
+ * @author Kai Kreuzer - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = ConsoleCommandExtension.class)
+public class NanoleafCommandExtension extends AbstractConsoleCommandExtension {
+
+    private static final String CMD_LAYOUT = "layout";
+    private final ThingRegistry thingRegistry;
+
+    @Activate
+    public NanoleafCommandExtension(@Reference ThingRegistry thingRegistry) {
+        super("nanoleaf", "Interact with the Nanoleaf integration.");
+        this.thingRegistry = thingRegistry;
+    }
+
+    @Override
+    public void execute(String[] args, Console console) {
+        if (args.length > 0) {
+            String subCommand = args[0];
+            switch (subCommand) {
+                case CMD_LAYOUT:
+                    if (args.length == 1) {
+                        thingRegistry.getAll().forEach(thing -> {
+                            if (thing.getUID().getBindingId().equals(NanoleafBindingConstants.BINDING_ID)) {
+                                ThingHandler handler = thing.getHandler();
+                                if (handler instanceof NanoleafControllerHandler) {
+                                    NanoleafControllerHandler nanoleafControllerHandler = (NanoleafControllerHandler) handler;
+                                    String layout = nanoleafControllerHandler.getLayout();
+                                    console.println("Layout of Nanoleaf controller '" + thing.getUID().getAsString()
+                                            + "' with label '" + thing.getLabel() + "':\n");
+                                    console.println(layout);
+                                    console.println("\n");
```

Comment:
Are you sure using `\n` here will work well across different OS-es?

Reply:
Good question. I have replaced it by `System.lineSeparator()`.